File: 단계별로 풀어보기/15. 약수, 배수와 소수 2/1934.py
def Euclidean(a, b):
    if b == 0:
        return a
    elif b == 1:
        return 1
    return Euclidean(b, a % b)


n = int(input())
for _ in range(n):
    num1, num2 = map(int, input().split())
    lcd = Euclidean(num1, num2) if num1 > num2 else Euclidean(num2, num1)
    result = int(num1 * num2 / lcd)
    print(result)

File: step by step/15. Measure and Drainage/1934.cpp
#include <iostream>
using namespace std;

int Euclidean(int a, int b)
{
	if (b == 0)
		return a;
	else if (b == 1)
		return 1;

	return Euclidean(b, a % b);
}

int main()
{
	int testCase;
	cin >> testCase;
	while (testCase--)
	{
		int num1, num2, LCD, result;
		cin >> num1 >> num2;
		LCD = (num1 > num2) ? Euclidean(num1, num2) : Euclidean(num2, num1);
		result = num1 * num2 / LCD;
		cout << result << endl;
	}
	return 0;
}